docs/developer-docs/gas-cost.md

```diff
@@ -1,100 +1,230 @@
 
-The following table shows the calculated storage cost per GB for a 30-day month: 
 
-|                      |                                    | 13-node Application Subnets | 34-node Application Subnets |
-|----------------------|------------------------------------|-----------------------------|-----------------------------|
-| GB Storage Per Month | For storing a GB of data per month | $0.431                      | $1.127                      |
+## Canister calls
+
+A canister call on ICP refers to invoking a function within a smart contract. Calls can come in the form of messages, executions, data storage, or they can use special features such as HTTPS outcalls, chain-key signing cryptography, and the Bitcoin API integration.
+
+At a high level, canister calls can be visualized using the following diagram:
+
+![Canister calls overview](./_attachments/transaction-overview.png)
+
+Each type of canister call has a different cycles cost associated with it, and the canister responsible for paying the cycles varies based on the type of call.
+
+### Canister creation
+
+An initial fee is charged when a canister is first created. Canisters are created by deploying a project onto an instance of the IC replica (locally or on the mainnet), or a canister can be created by another canister that is already running on a replica instance. Canisters created by another canister are referred to as 'child canisters'. The cycles charge for creating a canister is charged to the canister that is being created.
+
+### Messaging
+
+Messaging refers to calls that are made to a canister's methods. A canister's messaging costs depend on the type of message, the size of the message's request and response bytes, and the number of messages.
+
+Additional factors of cost include the number of nodes in the subnet the canister is deployed on, the total number of users interacting with the canister, and the number of daily active users.
+
+Messaging cost drivers include:
+
+- Message type:
+    - Query calls: Query data and perform a 'read-only' operation. (There is currently no cost for query calls).
+    - Update calls: Makes changes to the canister's state and must go through consensus.
+    - Inter-canister calls: Calls made between canisters.
+    - [Calls made by timers and heartbeats](/docs/current/developer-docs/backend/periodic-tasks): Periodic calls that are reoccurring.
+- Calls made daily per active user.
+- Size of each message's request bytes.
+- Instructions executed per message.
+- Calls per message.
+- Size of the message's request and response bytes per call.
+
+### Execution
+
+Execution refers to the tasks that the canister executes and the number of instructions executed per task and per call. These tasks use compute resources and rely on 'best-effort compute', meaning they do not pre-allocate compute resources unless otherwise specified, as compute resources are scarce and expensive.
```

Comment:
In order to handle an incoming message, the canister executes the function specified in the message. The execution cost consists of a fixed execution fee and per-instruction fee that is charged for each executed WebAssembly instructions: `base-fee + per-instruction-fee * number-of-instructions`. The current values of fees are `base-fee = 590K cycles (or $X)`, `per-instruction-fee = 0.4 cycles (or $X for 1B instructions)`.

By default canisters are scheduled for execution in a "best-effort" manner. Canisters that require guaranteed execution, can a share of compute capacity by setting `compute_allocation` in their canister settings. Compute allocation is expressed in percents and denote the percentage of an execution core reserved for the canister. For example, compute allocation of 50% means that the canister will get 50% of an execution core. In other words, it will be scheduled at least every other round. Compute allocation of 100% means that the canister will be scheduled every round. The current fee for 1 percent computer allocation per second is ...

Reply:
Revised to "In order to handle an incoming message, the canister executes the function specified in the message. The execution cost consists of a fixed execution fee and per-instruction fee that is charged for each executed WebAssembly instruction: `base-fee` + `per-instruction-fee` * `number-of-instructions`. The current values of fees are `base-fee` = 590K cycles (or $0.0000007885999 USD), `per-instruction-fee` = 0.4 cycles (or $0.000534644 USD for 1B instructions).

By default canisters are scheduled for execution in a "best-effort" manner. Canisters that require guaranteed execution can a share of compute capacity by setting `compute_allocation` in their canister settings. Compute allocation is expressed in percents and denote the percentage of an execution core reserved for the canister. For example, compute allocation of 50% means that the canister will get 50% of an execution core. In other words, it will be scheduled at least every other round. Compute allocation of 100% means that the canister will be scheduled every round. The current fee for 1 percent computer allocation per second is 10M cycles (or $0.0000133661 USD)."